Makefile

```diff
@@ -1,13 +1,17 @@
+# Pip needs to be < 25.1 for now because:
```

Comment:
Given how much else you're doing, would it make sense to split this up, do the uv conversion as one PR (which would maybe simplify things overall), and then do the crawlspace code changes as a different one?